### PR TITLE
Add basic documentation for debugging react native apps

### PR DESCRIPTION
**Pull request by francoislaberge**
I wasn't sure how a PR for a gh-page should be made. I kept it simple and just changed the files that generate the final website changes. I did not commit any generated files.

### CODE REVIEW

File: docs/NativeModulesIOS.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: Native Modules (iOS)
 layout: docs
 category: Guides
 permalink: docs/nativemodulesios.html
-next: testing
+next: debugging
 ---
```
Comment (francoislaberge) [inline docs/NativeModulesIOS.md]: I ordered Debugging before Testing as one was user facing, and one was for contributors. I figure, you will generally use react before contributing.

File: website/src/react-native/index.js [hunk 2]
```diff
@@ -66,6 +66,9 @@ var App = React.createClass({
           <p>
             All operations between the JavaScript application code and the native platform are performed asynchronously, and the native modules can also make use of additional threads as well.  This means we can decode images off of the main thread, save to disk in the background, measure text and compute layouts without blocking the UI, and more.  As a result, React Native apps are naturally fluid and responsive.  The communication is also fully serializable, which allows us to leverage Chrome Developer Tools to debug the JavaScript while running the complete app, either in the simulator or on a physical device.
           </p>
+          <p>
+            See <a href="docs/debugging.html">Debugging</a>.
+          </p>
           <img src="/react-native/img/chrome_breakpoint.png" width="800" height="443" />
```
Comment (francoislaberge) [inline website/src/react-native/index.js]: I added a link to the Debugging doc right after the debugging capability is pitched. I'm not attached to this change, but I think it's a nice touch.

File: docs/Debugging.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+---
+id: debugging
+title: Debugging
+layout: docs
+category: Guides
+permalink: docs/debugging.html
+next: testing
+---
+
+## Debugging React Native Apps
```
Comment (francoislaberge) [inline docs/Debugging.md]: A deeper explanation of how the debugging proxying works would be awesome, but beyond my current knowledge. I've kept it simple for now. I hope that someone will add more information here eventually.

**Comment by vjeux** [comment]
Merci! It's a great first draft, we'll improve it over time :)

**Comment by francoislaberge** [comment]
Awesome thanks. Noticed a small bug once it went live. I added a PR to fix it:
https://github.com/facebook/react-native/pull/424

**Comment by sahrens** [comment]
Good stuff, thanks for helping out!

**Comment by francoislaberge** [comment]
No problem. Awesome project guys, after trying to make PhoneGap/Cordova work for me over the years, I'm loving this approach.

**Comment by benjamingr** [comment]
Nice work +1 was definitely looking for this.

**Pull request merged**